commands/command_clone.go

```diff
@@ -79,8 +81,36 @@ func cloneCommand(cmd *cobra.Command, args []string) {
 		fetchRef("HEAD", include, exclude)
 	} else {
 		pull(include, exclude)
+
+		err := postCloneSubmodules(args)
+		if err != nil {
+			Exit("Error performing 'git lfs pull' for submodules: %v", err)
+		}
+	}
+
+}
+
+func postCloneSubmodules(args []string) error {
+	// In git 2.9+ the filter option will have been passed through to submodules
+	// So we need to lfs pull inside each, if the original clone command
+	if !git.Config.IsGitVersionAtLeast("2.9.0") {
+		// In earlier versions submodules would have used smudge filter
+		return nil
+	}
+	// Also we only do this if --recursive or --recurse-submodules was provided
+	if !cloneFlags.Recursive && !cloneFlags.RecurseSubmodules {
+		return nil
 	}
 
+	// Use `git submodule foreach --recursive` to cascade into nested submodules
+	// Also good to call a new instance of git-lfs rather than do things
+	// inside this instance, since that way we get a clean env in that subrepo
+	cmd := subprocess.ExecCommand("git", "submodule", "foreach", "--recursive",
```

Comment:
I agree, we probably don't need two independent layers of concurrency. If we did end up pursuing this, though, it would be easy to convince `git submodule foreach` to work with us. We could just have it run `pwd` in each submodule directory, and then apply the same code above to use that.

Either way, I think this is fine for now 😄